src/script/interpreter.cpp

```diff
@@ -1806,6 +1809,129 @@ bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript&
+                        return set_error(serror, SCRIPT_ERR_INTROSPECT_INDEX_OUT_OF_BOUNDS);
+                    const CTxOut out = outs->at(idx);
+
+                    switch (n)
+                    {
+                        case 0: // Asset
+                        {
+                            pushasset(stack, out.nAsset);
+                            break;
+                        }
+                        case 1: // Value
+                        {
+                            pushvalue(stack, out.nValue);
+                            break;
+                        }
+                        case 2: // Nonce
+                        {
+                            valtype vchOutNonce;
+                            if (out.nNonce.IsNull()) {
+                                stack.push_back(vchFalse);
+                            } else {
+                                vchOutNonce.insert(vchOutNonce.begin(), out.nNonce.vchCommitment.begin(), out.nNonce.vchCommitment.begin() + 33);
+                                stack.push_back(vchOutNonce);
+                            }
+                            break;
+                        }
+                        case 3: // spk
+                        {
+                            valtype vchScriptPubKey;
+                            vchScriptPubKey.insert(vchScriptPubKey.begin(), out.scriptPubKey.begin(), out.scriptPubKey.end());
+                            stack.push_back(vchScriptPubKey);
+                            break;
+                        }
+                        default:
+                            // Error on other values. This could be turned into OP_SUCCESS, but enforcing it
+                            // via SCRIPT_VERIFY_DISCOURAGE_OP_SUCCESS is not possible.
+                            // Alternatively, this can be turned into a OP_NOP, but restricts the use of this
+                            // opcode to read-only type.
+                            return set_error(serror, SCRIPT_ERR_INTROSPECT_INDEX_OUT_OF_BOUNDS);
+                    }
+                }
+                break;
+
+                case OP_INSPECTTX:
+                {
+                    // OP_INSPECTTX is available post tapscript
+                    if (sigversion == SigVersion::BASE || sigversion == SigVersion::WITNESS_V0) return set_error(serror, SCRIPT_ERR_BAD_OPCODE);
+
+                    if (stack.size() < 1)
+                        return set_error(serror, SCRIPT_ERR_INVALID_STACK_OPERATION);
+
+                    int n = CScriptNum(stacktop(-1), fRequireMinimal).getint();
+                    popstack(stack);
+
+                    switch (n)
+                    {
+                        case 0: // Version
+                        {
+                            // This does an inplicit conversion from version(int32_t) to (uint32_t)
+                            push4_le(stack, (uint32_t) checker.GetTxVersion());
```

Comment:
spelling: `implicit`.

Also it's not implicit, because you have an explicit cast :)

Reply:
I believe static_casts are preferred in C++.